packages/block-editor/src/components/block-tools/zoom-out-toolbar.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { dragHandle, trash } from '@wordpress/icons';
+import { Button, ToolbarButton } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as blocksStore } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import BlockDraggable from '../block-draggable';
+import BlockMover from '../block-mover';
+import Shuffle from '../block-toolbar/shuffle';
+import NavigableToolbar from '../navigable-toolbar';
+
+export default function ZoomOutToolbar( { clientId, rootClientId } ) {
+	const selected = useSelect(
+		( select ) => {
+			const {
+				getBlock,
+				hasBlockMovingClientId,
+				getNextBlockClientId,
+				getPreviousBlockClientId,
+				canRemoveBlock,
+				canMoveBlock,
+			} = select( blockEditorStore );
+			const { getBlockType } = select( blocksStore );
+			const { name } = getBlock( clientId );
+			const blockType = getBlockType( name );
+			const isBlockTemplatePart =
+				blockType?.name === 'core/template-part';
+
+			let isNextBlockTemplatePart = false;
+			const nextClientId = getNextBlockClientId();
+			if ( nextClientId ) {
+				const { name: nextName } = getBlock( nextClientId );
+				const nextBlockType = getBlockType( nextName );
+				isNextBlockTemplatePart =
+					nextBlockType?.name === 'core/template-part';
+			}
+
+			let isPrevBlockTemplatePart = false;
+			const prevClientId = getPreviousBlockClientId();
+			if ( prevClientId ) {
+				const { name: prevName } = getBlock( prevClientId );
+				const prevBlockType = getBlockType( prevName );
+				isPrevBlockTemplatePart =
+					prevBlockType?.name === 'core/template-part';
+			}
+
+			return {
+				blockMovingMode: hasBlockMovingClientId(),
+				isBlockTemplatePart,
+				isNextBlockTemplatePart,
+				isPrevBlockTemplatePart,
+				canRemove: canRemoveBlock( clientId, rootClientId ),
+				canMove: canMoveBlock( clientId, rootClientId ),
+			};
+		},
+		[ clientId, rootClientId ]
+	);
+
+	const {
+		blockMovingMode,
+		isBlockTemplatePart,
+		isNextBlockTemplatePart,
+		isPrevBlockTemplatePart,
+		canRemove,
+		canMove,
+	} = selected;
+
+	const { removeBlock } = useDispatch( blockEditorStore );
+
+	const classNames = clsx( 'zoom-out-toolbar', {
```

Comment:
I wonder if `vertical-toolbar` is a better name as this could be used in other context.